connectors/src/resources/zendesk_resources.ts

```diff
@@ -98,23 +99,43 @@ export class ZendeskBrandResource extends BaseResource<ZendeskBrand> {
    */
   async delete(transaction?: Transaction): Promise<Result<undefined, Error>> {
     await this.model.destroy({
-      where: {
-        connectorId: this.connectorId,
-      },
+      where: { connectorId: this.connectorId, brandId: this.brandId },
       transaction,
     });
     return new Ok(undefined);
   }
 
   /**
-   * Deletes the brand and all associated data (tickets, categories, articles).
+   * Deletes all the data stored in the db and in the data source relative to a brand (category, articles and tickets).
    */
-  // eslint-disable-next-line no-empty-pattern
-  async remove({}: {
+  async deleteWithChildren({
+    dataSourceConfig,
+  }: {
     dataSourceConfig: DataSourceConfig;
-    loggerArgs: Record<string, string | number>;
   }) {
-    // TODO: implement removal for the children first.
+    const categories = await ZendeskBrandResource.fetchAllCategories({
+      connectorId: this.connectorId,
+      brandId: this.brandId,
+    });
+    const tickets = await ZendeskBrandResource.fetchAllTickets({
+      connectorId: this.connectorId,
+      brandId: this.brandId,
+    });
+    await Promise.all([
+      ...categories.map((category) =>
+        category.deleteWithChildren({ dataSourceConfig })
```

Comment:
Could we have a method on ZendeskCategoryResource that deletes all from a brandId? We would do one query instead of n 🙏🏻

Reply:
done!